### PR TITLE
[console] Start migration of Api portal details screen

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

https://github.com/gravitee-io/issues/issues/8141

## Description

Start migration of Api portal details screen

## Additional context
image

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/8141-migrate-api-details/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-ihrkzxednx.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    expect(req.request.body.picture).toEqual('data:image/png;base64,');
+    expect(req.request.body.background).toEqual('data:image/png;base64,');
+    expect(req.request.body.labels).toEqual(['label1', 'label2', 'label3']);
+    expect(req.request.body.categories).toEqual(['category1', 'category2']);
+  });
+
+  function expectApiGetRequest(api: Api) {
+    httpTestingController.expectOne({ url: `${CONSTANTS_TESTING.env.baseURL}/apis/${api.id}`, method: 'GET' }).flush(api);
+    fixture.detectChanges();
+  }
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.spec.ts:141]: I'm thinking about doing a pull request to externalise this kind of small functions in a "api-service.testing.ts"

File: gravitee-apim-console-webui/src/management/api/portal/apis.portal.route.ts [hunk 2]
```diff
@@ -54,6 +54,20 @@ function apisPortalRouterConfig($stateProvider) {
...
+      component: 'ngApiPortalDetails',
+      data: {
+        useAngularMaterial: true,
+        // menu: {
+        //   label: 'Portal',
+        //   icon: 'important_devices',
+        // },
+        // docs: {
+        //   page: 'management-api',
+        // },
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/portal/apis.portal.route.ts]: Maybe we can uncomment it or remove it for the moment ?

File: gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.html [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      <div class="details-card__header__info-inputs">
+        <div class="details-card__header__info-inputs__first-row">
+          <mat-form-field class="details-card__header__info-inputs__first-row__name-field" appearance="fill">
+            <mat-label>Name</mat-label>
+            <input #apiName matInput formControlName="name" required />
+          </mat-form-field>
+
+          <mat-form-field class="details-card__header__info-inputs__first-row__version-field" appearance="fill">
+            <mat-label>Version</mat-label>
+            <input matInput formControlName="version" required />
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.html:33]: I would expect to have the required in the form control. Imho it's clearer to have everything in it, initialise the component with the form group, the form controls and their validators on each one. Wdyt ?

Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.html:33]: Should we add the errors messages on the required fields ?

File: gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.html [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                Drag your image
+              </span>
+              <gio-avatar
+                class="details-card__header__right-coll_media__picture__btn__default-avatar"
+                size="114"
+                [name]="apiName.value"
+              ></gio-avatar>
+            </gio-form-file-picker-add-button>
+          </gio-form-file-picker>
+          <gio-form-file-picker class="details-card__header__right-coll__media__background" formControlName="background" accept="image/*">
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.html:79]: The form is dirty when we click on the file picker even if I click on cancel directly. So we have the save bar even if there is no changes. Maybe this is something we can think about. Not here but in a near future. We can try to improve the file picker 🤷

**Review by Okhelifi — APPROVED** [review]
LGTM

File: gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.spec.ts (hunk 1 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.spec.ts:141]: yes 👌

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.spec.ts:141]: difficult if the detectChanges is needed 🤷‍♂️ but not sure if it is

File: gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.html (hunk 4 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/details/api-portal-details.component.html:79]: i will see later if i can fix this. fist try to finish this page ...

**Pull request merged**